app/models/dfe_sign_in_user.rb

```diff
@@ -36,11 +53,28 @@ def self.load_from_session(session)
       dfe_sign_in_uid: dfe_sign_in_session['dfe_sign_in_uid'],
       first_name: dfe_sign_in_session['first_name'],
       last_name: dfe_sign_in_session['last_name'],
+      id_token: dfe_sign_in_session['id_token'],
     )
   end
 
   def self.end_session!(session)
     session.delete('post_dfe_sign_in_path')
     session.delete('dfe_sign_in_user')
   end
+
+private
+
+  # a URL the user can visit to log them out of DSI and be redirected to our
+  # after-sign-out path where we'll delete their local session
+  def dsi_logout_url(interface:)
+    logout_url = URI.parse("#{ENV.fetch('DFE_SIGN_IN_ISSUER')}/session/end").tap do |url|
```

Comment:
Not important but I think this can be visually simplified:

```rb
query = {
  post_logout_redirect_uri: auth_dfe_sign_out_url,
  id_token_hint: @id_token,
  state: interface,
}

"#{ENV.fetch('DFE_SIGN_IN_ISSUER')}/session/end?#{query.to_query}"
```